File: 20242Q/poo/laboratorio/tp06/ej4/src/CheckingAccount.java
public class CheckingAccount extends BankAccount{

    private double overdraft;

    public CheckingAccount(int id, double overdraft) {
        super(id);
        this.overdraft = overdraft;
    }

    protected boolean canExtract(double amount) {
        if(balance - amount <= overdraft) {
            return true;
        }

        return false;
    }

    // esto no se usa, pero lo implemento igual
    public double getOverdraft() {
        return overdraft;
    }
}

// copiado

File: 20242Q/poo/laboratorio/tp08/ej5/checking_account.rb
require_relative 'bank_account'

class CheckingAccount < BankAccount

  def initialize(id, overdraft)
    initialize_bank_account(id)
    @overdraft = overdraft
  end

  def extract?(amount)
    @balance + @overdraft - amount >= 0
  end

  private :extract?
end